core/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala

```diff
@@ -53,7 +53,10 @@ private[spark] abstract class YarnSchedulerBackend(
    * This includes executors already pending or running.
    */
   override def doRequestTotalExecutors(requestedTotal: Int): Boolean = {
-    yarnSchedulerEndpoint.askWithRetry[Boolean](RequestExecutors(requestedTotal))
+    if (conf.getBoolean("spark.dynamicAllocation.enabled", false)) {
```

Comment:
You mean:

    private[spark] override def requestTotalExecutors

I don't see any calls to it, and given it's `private[spark]`...